### ISSUE TITLE
Sun in clear sky conditions prevent the menu icons from being visible when dark mode is forced and device is tilted

### ISSUE THREAD

**Issue by jawbu** [breezy-weather/breezy-weather#2879]
### Steps to reproduce

1. Open app with a weather source
2. Select any place with heat (for light background, e.g. clear sky)
3. (With gravity enabled) slightly tilt phone
4. Option symbol colours conflict with background colours

### Expected behavior

Clear, contrasting visuals, no matter the background, for accessing upper main screen options

### Actual behavior

Because the upper main screen options have no outlines/bounding boxes in quite a few cases (usually if the orientation is slightly off) they are obscured, leading to many miss clicks.

<img width="653" height="901" alt="Image" src="https://github.com/user-attachments/assets/ab65bb6f-640a-44e8-81bf-da21d74b6f7e" />

### Weather source used

Any weather

### Breezy Weather version

6.2.1

### Android version

Android 17

### Device

Pixel 8

### Other details

While you can simply change the colours of the background, this is just treating the symptoms and not the cause, I would really recommend putting the icons in semitransparent pills, or adding a short, slightly dark downward gradient from the top, so you avoid these issues entirely.

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [x] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [x] I have updated the app to **[version 6.2.1 or later](https://github.com/breezy-weather/breezy-weather/releases)** and I can reproduce the issue in this version.
- [x] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [x] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> Because the upper main screen options have no outlines/bounding boxes in quite a few cases (usually if the orientation is slightly off) they are obscured, leading to many miss clicks.

This may sound a bit stupid, but have you thought of tilting your device to see the menu icons correctly, if you are having that many miss clicks?

I noticed that issue during development, but I thought it was pretty obvious that just tilting a bit made it easier to read, so I didn't think it required a particular fix. We could lower the sun a bit, but a sun in the middle of the screen feels a bit out of the place.

> I would really recommend putting the icons in semitransparent pills

I never saw semitransparents pills in Material Design. It looks like you can create buttons, but this would be too much IMO to have all icons like that.

> or adding a short, slightly dark downward gradient from the top, so you avoid these issues entirely.

This won't prevent the sun from going on top of that new gradient.


Alternate workaround to consider if tilting your device is not possible: Settings > Appearance > Location-based dark mode = Follow day/night. This will prevent day conditions which are usually bright from being displayed in dark mode.


If none of these options suit you, we can look into making a dark sun, but this feels even more apocalyptic IMO:
<img width="300" alt="Screenshot of the main screen with a black sun" src="https://github.com/user-attachments/assets/c28157cd-8005-47be-893b-554544f76b53" />

**Comment by jawbu** [comment]
> have you thought of tilting your device to see the menu icons correctly, if you are having that many miss clicks?

Whenever casually lying down, having to rotate the phone around correctly in order to see options correctly feels counterproductive. Even so, having any weak points that impede the sight of the options is worth fixing.

> I never saw semitransparents pills in Material Design. It looks like you can create buttons, but this would be too much IMO to have all icons like that.

I've come across it quite a lot? e.g. AOSP quick settings:
<img width="1080" height="276" alt="Image" src="https://github.com/user-attachments/assets/2e13e357-4b38-444c-babf-8a1c85acf33c" />

> > or adding a short, slightly dark downward gradient from the top, so you avoid these issues entirely.

> This won't prevent the sun from going on top of that new gradient.

I meant static with the UI elements, not as part of the weather backgrounds, e.g. Google Gallery:

<img width="1080" height="439" alt="Image" src="https://github.com/user-attachments/assets/a24fd7dc-0e57-43e5-8ff0-90a166bc76ec" />

> Alternate workaround to consider if tilting your device is not possible: Settings > Appearance > Location-based dark mode = Follow day/night. This will prevent day conditions which are usually bright from being displayed in dark mode.

This does solve it, but in a visibly inconsistent manner, switching between different themes wherever the wind blows

**Comment by papjul** [comment]
> Whenever casually lying down, having to rotate the phone around correctly in order to see options correctly feels counterproductive.

Aaah, I understand why you're the only one having the issue now: I can't reproduce this issue when lying down, as the sun clearly never go that up when the phone is either vertical or facing the ground.

I have no idea what's going on with your sensor, but you can turn sensor effects off in the app. Settings > Appearance > Gravity sensor. This will ensure the sun sticks in a position that doesn't affect readability.

**Comment by jawbu** [comment]
How would this be a device sensor issue? This is the app incorrectly stopping the bounds of the movement?

I have had these visibility issues on multiple devices

**Comment by papjul** [comment]
> How would this be a device sensor issue?

The sun is supposed to go down when the device is hold vertically or facing the ground, not up (like on your screenshot).

You should never have this visibility issue when lying down.

> This is the app incorrectly stopping the bounds of the movement?

It is a good idea to look into making the sun stopping earlier its movement to the top. I will see if it is feasible.

**Comment by glen-nicol** [comment]
I have also run into this recently. My natural screen tilt angle while standing or sitting places the sun right underneath the icons. I think the default advice to tilt the screen to see important menu items is an accessibility problem. Imagine telling a senior citizen how to remove a location that's always sunny.

Would it be too much to add a slight gradient to the entire top of the page since all that text and iconography is white?

Android itself does this for the navigation buttons.

**Comment by papjul** [comment]
> I think the default advice to tilt the screen to see important menu items is an accessibility problem.

There are other workarounds:
- Turn off gravity sensor
- Revert back to location-based dark mode
- Turn off backgrounds
- Wait for the night

But yes, I agree, it's an accessibility problem.

> Imagine telling a senior citizen how to remove a location that's always sunny.

If that senior was able to find how to force dark mode during the day, they are skilled enough to find out about one of the workarounds. Also, unless you live in the far North or far South at a specific time of the year, a location is not "always" sunny.

> Would it be too much to add a slight gradient to the entire top of the page since all that text and iconography is white?
> 
> Android itself does this for the navigation buttons.

If you're talking about the background shade, no modern app does that for the top bar. Also, it does not guarantee it will be enough to have a decent contrast that matches accessibility guidelines.

Currently, the options being studied are:
- Make the sun stop earlier its movement to the top, but it will likely be the same as turning off the gravity sensor, which you can already do
- Make the sun black

It doesn't make sense to have such a big bright item on screen when the user explicitly requests dark mode to ease their eyes. The first option does not solve that issue, so the latter is what makes the most sense to me so it solves two accessibility issues at once.
